core/src/main/java/com/linecorp/armeria/client/retry/RetryRule.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import java.util.concurrent.CompletionStage;
+import java.util.function.Predicate;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.UnprocessedRequestException;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.HttpStatusClass;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.common.ResponseHeaders;
+
+/**
+ * Determines whether a failed request should be retried.
+ * If you need to determine by looking into the {@link Response}, use {@link RetryRuleWithContent}.
+ */
+@FunctionalInterface
+public interface RetryRule {
+
+    /**
+     * Returns a newly created {@link RetryRuleBuilder}.
+     */
+    static RetryRuleBuilder builder() {
+        return new RetryRuleBuilder();
+    }
+
+    /**
+     * Returns a newly created {@link RetryRule} that will retry with the specified {@link Backoff}
```

Comment:
`default backoff`

Reply:
Few of those throughout. Maybe the javadoc copy-paste errors are an indication of too many factories? :) Would consider not having the ones that take backoff, using the builder hardly adds many characters but makes the API much easier to read through on a javadoc for example.

Reply:
Yeah, `RetryRule` has too many factories. Let me remove factories that take backoff.